### PR TITLE
Added status bar to image viewer

### PR DESCRIPTION
**Pull request by hajdam**
More changes to image viewer
- Added optional status bar with zoom level and image resolution information
- Added configurations for general viewer/editor and image viewer
- Added initial zoom mode: native resolution / resize window / fit to window
- Added viewer/editor window position and fullscreen state saving
- Fixes issue with right click zoom without Ctrl key

### CODE REVIEW

File: mucommander-core/src/main/java/com/mucommander/ui/viewer/FileEditorPresenter.java [hunk 1]
```diff
@@ -152,24 +154,32 @@ public void windowClosing(WindowEvent e) {
...
-                        getFrame().setFullScreen(!fullScreen);
-                        fullScreenMenuItem.setSelected(!fullScreen);
-                    }
-            );
-            fullScreenMenuItem.setSelected(getFrame().isFullScreen());
+                        switchFullScreenMode(!fullScreen);
+                    });
+
+            String windowWidthValue = EditorPreferences.WINDOW_WIDTH.getValue();
+            if (!"".equals(windowWidthValue)) {
```
Comment (ahadas) [inline mucommander-core/src/main/java/com/mucommander/ui/viewer/FileEditorPresenter.java:161]: how about
```suggestion
            if (!windowWidthValue.isEmpty()) {
```
we don't allow null for EditorPreferences.WINDOW_WIDTH, right?

File: mucommander-core/src/main/java/com/mucommander/ui/viewer/FileViewerPresenter.java [hunk 2]
```diff
@@ -161,27 +175,50 @@ protected void show(AbstractFile file) throws IOException {
...
-                        getFrame().setFullScreen(!fullScreen);
-                        fullScreenMenuItem.setSelected(!fullScreen);
-                    }
-            );
-            fullScreenMenuItem.setSelected(getFrame().isFullScreen());
+                        switchFullScreenMode(!fullScreen);
+                    });
+
+            String windowWidthValue = ViewerPreferences.WINDOW_WIDTH.getValue();
+            if (!"".equals(windowWidthValue)) {
```
Comment (ahadas) [inline mucommander-core/src/main/java/com/mucommander/ui/viewer/FileViewerPresenter.java:182]: the comment on file editor presenter applies here as well

File: mucommander-core/src/main/java/com/mucommander/ui/viewer/FileViewerPresenter.java [hunk 3]
```diff
@@ -161,27 +175,50 @@ protected void show(AbstractFile file) throws IOException {
...
             viewerMenu.add(fullScreenMenuItem);
 
             closeMenuItem = MenuToolkit.addMenuItem(viewerMenu,
                     Translator.get("file_viewer.close"),
                     menuItemMnemonicHelper,
                     KeyStroke.getKeyStroke(KeyEvent.VK_ESCAPE, 0),
                     (e) -> {
-                        fileViewer.close();
-                        getFrame().dispose();
+                        getFrame().dispatchEvent(new WindowEvent(getFrame(), WindowEvent.WINDOW_CLOSING));
```
Comment (ahadas) [inline mucommander-core/src/main/java/com/mucommander/ui/viewer/FileViewerPresenter.java:203]: we can simplify it both here and in the viewer then with:
e -> getFrame().dispatchEvent(new WindowEvent(getFrame(), WindowEvent.WINDOW_CLOSING))

File: mucommander-translator/src/main/resources/dictionary.properties [hunk 4]
```diff
@@ -566,6 +566,10 @@ file_viewer.close = Close
 file_viewer.large_file_warning = This file may be too large for this operation.
 file_viewer.open_anyway = Open anyway
 file_viewer.fullscreen = Full screen
+file_viewer.window_position_x = Window position X
+file_viewer.window_position_y = Window position Y
+file_viewer.window_width = Window width
+file_viewer.window_height = Window height
```
Comment (ahadas) [inline mucommander-translator/src/main/resources/dictionary.properties:572]: where would these values be presented?

File: mucommander-translator/src/main/resources/dictionary.properties [hunk 5]
```diff
@@ -598,6 +602,11 @@ text_viewer.tab_size = Tab size
 text_viewer.whitespace_visible = Whitespace visible
 image_viewer.view_menu = $[text_viewer.view]
 image_viewer.controls_menu = Controls
+image_viewer.view_status_bar = Status Bar
+image_viewer.initial_zoom_menu = Initial Zoom
+image_viewer.initial_zoom_as_native = Native Size
+image_viewer.initial_zoom_fit_to_view = Fit to View
+image_viewer.initial_zoom_resize_window = Resize Window
```
Comment (ahadas) [inline mucommander-translator/src/main/resources/dictionary.properties:609]: besides 'Select All' in all other properties I see only the first word with a capital letter - let's keep that convention?

File: mucommander-viewer-image/src/main/java/com/mucommander/viewer/image/ui/ImageStatusPanel.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        initComponents();
+    }
+
+    private void initComponents() {
+        imageSizeLabel = new javax.swing.JLabel();
+        zoomLabel = new javax.swing.JLabel();
+
+        imageSizeLabel.setHorizontalAlignment(javax.swing.SwingConstants.CENTER);
+        imageSizeLabel.setText("-");
+        imageSizeLabel.setToolTipText(Translator.get("binary_viewer.status.documentSizeLabel.toolTipText"));
```
Comment (ahadas) [inline mucommander-viewer-image/src/main/java/com/mucommander/viewer/image/ui/ImageStatusPanel.java:42]: wrong indentation?

File: mucommander-viewer-image/src/main/java/com/mucommander/viewer/image/ui/ImageViewerPanel.java [hunk 7]
```diff
@@ -0,0 +1,125 @@
+package com.mucommander.viewer.image.ui;
```
Comment (ahadas) [inline mucommander-viewer-image/src/main/java/com/mucommander/viewer/image/ui/ImageViewerPanel.java:1]: please add the top level comment

**Review by ahadas — COMMENTED** [review]
@hajdam thanks for posting this - nice changes!
the comments inside are pretty minor but there's a bigger question that also appears in the issue I correlated this PR with - about the granularity of the saved changes. this PR saves a single position for all editors and viewers - if I used both text editor and text viewer, the configuration of which component will be saved? and I also see that the position of image viewer is not persisted..

File: mucommander-translator/src/main/resources/dictionary.properties (hunk 4 above)
Comment (hajdam) [reply mucommander-translator/src/main/resources/dictionary.properties:572]: I followed "template" from TextViewerPreferences, but it's probably not needed...

**Comment by hajdam** [comment]
Window position is saved on Viewer or Editor Presenter closing. I don't mind either way - could be removed or whatever...

**Comment by ahadas** [comment]
> Window position is saved on Viewer or Editor Presenter closing. I don't mind either way - could be removed or whatever...

ah right, that explains what I saw - I closed the editor after closing the viewer..

**Review by ahadas — APPROVED** [review]
I think it would be better to store the position of the window per viewer or editor type but even without that, this PR improves the current state so let's get it in. thanks @hajdam !

**Pull request merged**